### DISCUSSION TITLE
Sqids support in ClickHouse

### DISCUSSION THREAD

**Discussion by rschu1ze** [sqids/sqids-spec#18] [Show and tell]
Hi,

I could not find a repository for the [website](https://sqids.org) on GitHub (and could not make a pull request) so I figured I could open this thread instead.

[ClickHouse](https://clickhouse.com/) (a popular SQL database for real-time analytics) recently added a SQL function `sqid()` ([docs](https://clickhouse.com/docs/en/sql-reference/functions/hash-functions#sqid), [PR](https://github.com/ClickHouse/ClickHouse/pull/57512)) which (not surprisingly) generate sqids from SQL, e.g. `SELECT sqid(125, 126)`. If you feel it is appropriate, you might want to add ClickHouse to a section "Used in ..." on the website, similar to what the [website for roaring bitmaps](https://roaringbitmap.org/) does.

**Comment by 4kimov** [comment]
Hi @rschu1ze,

Thank you for the heads up. Yes, I saw that mentioned in ClickHouse's CHANGELOG a few days ago, pretty cool. We don't have a dedicated "used in _other projects_" section, but we do have a list of repos/code, so let me see how that can be added there (it has a particular structure that would have to be adjusted).

In the meantime, noticed a minor tidbit on the ClickHouse docs, where it says:

> Returned Value: A hash id String.

Technically, Sqids encoded numbers are not a _hash_ since the output can be decoded back into numbers, so we just refer to the output as IDs (or strings). Just thought it's worth a mention.

I'll leave this discussion open until ClickHouse is added to the website.

**Comment by rschu1ze** [comment]
Sure, thanks.

You are right about the docs, I fixed the wording here: https://github.com/ClickHouse/ClickHouse/pull/58517

**Comment by 4kimov** [comment]
Hi @rschu1ze,

Cool, thank you for adjusting the wording. ClickHouse page has been added here: https://sqids.org/clickhouse Do you see anything that should be improved or communicated better?

**Comment by rschu1ze** [comment]
Hi @4kimov,

very nice, thanks! The only thing you might want to improve is to add the ClickHouse logo (see https://github.com/ClickHouse/) here:

<img width="223" alt="image" src="https://github.com/sqids/sqids-spec/assets/100446237/8c4ea8d1-65a6-4525-aed5-3519de9a936d">

**Comment by 4kimov** [comment]
Hi @rschu1ze,

Yeah, we are missing icons for a few languages - I'm going to try to address that within the next few releases. Thank you for the feedback, closing this one as done.

**Reply by 4kimov** [reply]
Icon added.

**Comment by mgred** [comment]
@rschu1ze thanks for you work - awesome :heart: 

Some thoughts that come to my mind: In other sqids projects, we use to have some function/struct/object exported named `sqids` or `Sqids` (plural) that gives methods to `encode` and `decode`.

Questions:
- Should the function better be named `sqids`?
- the `sqid` function in ClickHouse _only_ encodes. Would there be a usecase to also decode?

**Reply by rschu1ze** [reply]
@mgred Thanks for the feedback.

By convention, SQL function names (at least in ClickHouse) are in singular, e.g. [here](https://clickhouse.com/docs/en/sql-reference/functions/hash-functions) there is `farmHash64()` but not `farmHashes64()`.

About decoding, I added that in this [PR](https://github.com/ClickHouse/ClickHouse/pull/58544).

**Reply by mgred** [reply]
Ah, nice! And you renamed to `encodeSqid`, I like that - makes it more specific :+1: Personally, I would go with `sqidsEncode`, just to follow the pattern, but have no clue about naming in ClickHouse and best practice.

Again thanks for sharing

BTW, I left a little comment :wink:

**Reply by rschu1ze** [reply]
Cool, thanks again. You have a point about the word order, I switched to `sqidEncode()` and `sqidDecode()` (still using singular).

**Reply by rschu1ze** [reply]
@4kimov As https://github.com/ClickHouse/ClickHouse/pull/58544 rolls in, you probably want to change the example on https://sqids.org/clickhouse to

```sql
SELECT sqidEncode(1, 2, 3); -- '86Rf07'
SELECT sqidDecode('86Rf07)'; -- [1, 2, 3]
```

**Reply by 4kimov** [reply]
@rschu1ze Copy that. Subscribed to the PR, will update the site when this change goes out. Thanks.
